File: android-lib/src/androidTest/java/com/kinvey/androidTest/callback/CustomKinveyClientCallback.java
package com.kinvey.androidTest.callback;

import com.kinvey.java.core.KinveyClientCallback;

import java.util.concurrent.CountDownLatch;

/**
 * Created by yuliya on 09/20/17.
 */

public class CustomKinveyClientCallback<T> implements KinveyClientCallback<T> {

    private CountDownLatch latch;
    private T result;
    private Throwable error;

    public CustomKinveyClientCallback(CountDownLatch latch) {
        this.latch = latch;
    }

    @Override
    public void onSuccess(T result) {
        this.result = result;
        finish();
    }

    @Override
    public void onFailure(Throwable error) {
        this.error = error;
        finish();
    }

    void finish() {
        latch.countDown();
    }

    public T getResult() {
        return result;
    }

    public Throwable getError() {
        return error;
    }
}

File: android-lib/src/androidTest/java/com/kinvey/androidTest/callback/CustomKinveyClientCallback.kt
package com.kinvey.androidTest.callback

import com.kinvey.java.core.KinveyClientCallback
import java.util.concurrent.CountDownLatch

/**
 * Created by yuliya on 09/20/17.
 */

class CustomKinveyClientCallback<T>(private val latch: CountDownLatch) : KinveyClientCallback<T> {
    var result: T? = null
        private set
    var error: Throwable? = null
        private set

    override fun onSuccess(result: T?) {
        this.result = result
        finish()
    }

    override fun onFailure(error: Throwable?) {
        this.error = error
        finish()
    }

    internal fun finish() {
        latch.countDown()
    }

}